massa-models/src/operation.rs

```diff
@@ -35,6 +37,55 @@ pub struct OperationPrefixId(Vec<u8>);
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
 pub struct OperationSuffixId(Vec<u8>);
 
+/// Manage the relation between [OperationPrefixId] and [OperationId]
+///
+/// note: we could think about replace `Vec<OperationId>` with `Vec<OperationSuffixId>`
+///       if the execution time CPU is equivalent
+#[derive(Default)]
+pub struct OperationIdAdapter(Map<OperationPrefixId, OperationIds>);
```

Comment:
Since this is only, and likely only to be, used in `protocol`, I think it belongs there and should remain hidden from the rest of the codebase. 

Also, in the light of my other comment, I think it can be renamed something along the lines of `CheckedOperations`